builtin/logical/transit/backend.go

```diff
@@ -225,15 +225,20 @@ func (b *backend) autoRotateKeys(ctx context.Context, req *logical.Request) erro
 			errs = multierror.Append(errs, err)
 			continue
 		}
+		if !b.System().CachingDisabled() {
+			p.Lock(true)
+		}
 
 		// If the policy is nil, move onto the next one.
 		if p == nil {
+			p.Unlock()
```

Comment:
Won't this and the other unlock calls throw if caching is enabled?

Reply:
Ah, I see now that the behaviour is as in the previous code.

Reply:
Well that one in particular will panic because p is nil. Other unlock calls shouldn't panic because when caching is enabled we manually lock the policy and when caching is disabled GetPolicy returns the policy already locked.